### PR TITLE
[4.0] http headers language

### PR DESCRIPTION
**Pull request by brianteeman**
Small tweaks to the language file.

@zero-24 we could make the strings even simpler and the tool easier to use if the HTTP-Header field was a select of the supported values instead of a free text field - thoughts?

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
...r><ul><li><a href='https://scotthelme.co.uk/hardening-your-http-response-headers/#x-frame-options'>'X-Frame-Options: SAMEORIGIN'</a></li><li><a href='https://scotthelme.co.uk/hardening-your-http-esponse-headers/#x-xss-protection'>'X-XSS-Protection: 1; mode=block'</a></li><li><a href='https://scotthelme.co.uk/hardening-your-http-response-headers/#x-content-type-options'>'X-Content-Type-Options: nosniff'</a></li><li><a href='https://scotthelme.co.uk/a-new-security-header-referrer-policy/'>'Referrer-Policy: no-referrer-when-downgrade'</a></li></ul><br>The full list of supported headers are: <br><ul><li><a href='https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security'>Strict-Transport-Security</a></li><li><a href='https://en.wikipedia.org/wiki/Content_Security_Policy'>Content-Security-Policy</a></li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul><br>These headers help your browser to protect your website from <a href='https://en.wikipedia.org/wiki/Cross-site_scripting'>XSS</a> and <a href='https://en.wikipedia.org/wiki/Clickjacking'>Clickjacking</a> attacks."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini]: Missing `r` in `esponse` in `https://scotthelme.co.uk/hardening-your-http-esponse-headers/#x-xss-protection`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d9c194012899abb3f405a4dfe5b52932060279d1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20459">issues.joomla.org/tracker/joomla-cms/20459</a>.</sub>

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 2]
```diff
@@ -6,8 +6,11 @@
...
 PLG_SYSTEM_HTTPHEADERS="System - HTTPHeaders"
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Define additional HTTP Header"
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="Using this you can set also different values from the default ones and also additional headers when it is supported. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_XML_DESCRIPTION="This Plugin can set some Security HTTP Headers"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Customise the HTTP Headers"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="Replace the default headers with your own custom values and add additional headers. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_CLIENT="Client"
```
Comment (zero-24) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini:12]: hmm this string is not used any more? Why do you want to add it back? It is now `PLG_SYSTEM_HTTPHEADERS_HEADER_CLIENT` ;)

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 3]
```diff
@@ -6,8 +6,11 @@
...
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="Using this you can set also different values from the default ones and also additional headers when it is supported. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_XML_DESCRIPTION="This Plugin can set some Security HTTP Headers"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Customise the HTTP Headers"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="Replace the default headers with your own custom values and add additional headers. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_CLIENT="Client"
+PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_CLIENT_BOTH="Both"
```
Comment (zero-24) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini:13]: Also this string is not used anymore as it is now: `PLG_SYSTEM_HTTPHEADERS_HEADER_CLIENT_BOTH`

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 4]
```diff
@@ -6,8 +6,11 @@
 COM_PLUGINS_HSTS_FIELDSET_LABEL="Strict-Transport-Security (HSTS)"
 COM_PLUGINS_CSP_FIELDSET_LABEL="Content-Security-Policy (CSP)"
 PLG_SYSTEM_HTTPHEADERS="System - HTTPHeaders"
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER="Define additional HTTP Header"
-PLG_SYSTEM_HTTPHEADERS_ADDITIONAL_HEADER_DESC="Using this you can set also different values from the default ones and also additional headers when it is supported. The supported headers are: <br><ul><li>Strict-Transport-Security</li><li>Content-Security-Policy</li><li>Content-Security-Policy-Report-Only</li><li>X-Frame-Options</li><li>X-XSS-Protection</li><li>X-Content-Type-Options</li><li>Referrer-Policy</li><li>Expect-CT</li></ul>"
+PLG_SYSTEM_HTTPHEADERS_XML_DESCRIPTION="This Plugin can set some Security HTTP Headers"
```
Comment (zero-24) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini:9]: This string is translated in line 46 why do we want to add it here too ;)

**Comment by zero-24** [comment]
> we could make the strings even simpler and the tool easier to use if the HTTP-Header field was a select of the supported values instead of a free text field - thoughts?

Yes this could be possible. But I would suggest to do this in a separate PR as this also requires PHP code to change ;)

**Review by zero-24 — CHANGES_REQUESTED** [review]
please see my comments

**Comment by brianteeman** [comment]
Obviously made an error doing this from my phone. Closing now and will redo it later or if you think the select is a good idea then I can do new strings after/with that. We don't need to list supported headers in the text any more then as it will be obvious from the select

**Comment by zero-24** [comment]
Ok I'm going to propose a PR. So we can include the language changes there too.

**Comment by brianteeman** [comment]
Great

**Comment by zero-24** [comment]
Done: https://github.com/joomla/joomla-cms/pull/20491 just let me know what you want to have in there concerning the language strings.

**Comment by brianteeman** [comment]
Great. I Will

**Pull request closed without merging**